library/modules/bigip_profile_http.py

```diff
@@ -0,0 +1,670 @@
+    user: admin
+    password: secret
+  delegate_to: localhost
+
+- name: Add HTTP profile for transparent proxy
+  bigip_profile_http:
+    name: my_profile
+    server: lb.mydomain.com
+    user: admin
+    proxy_type: transparent
+    password: secret
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+description:
+  description: Description of the profile.
+  returned: changed
+  type: string
+  sample: My profile
+proxy_type:
+  description: Specify proxy mode of the profile.
+  returned: changed
+  type: string
+  sample: explicit
+insert_xforwarded_for:
+  description: Insert X-Forwarded-For-Header.
+  returned: changed
+  type: bool
+  sample: yes
+redirect_rewrite:
+  description: Rewrite URI that are part of 3xx responses.
+  returned: changed
+  type: string
+  sample: all
+encrypt_cookies:
+  description: Cookie names to encrypt.
+  returned: changed
+  type: list
+  sample: ['MyCookie1', 'MyCookie2']
+dns_resolver:
+  description: Configured dns resolver.
+  returned: changed
+  type: string
+  sample: '/Common/FooBar'
+'''
+import q
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import env_fallback
+
+try:
+    from library.module_utils.network.f5.bigip import F5RestClient
+    from library.module_utils.network.f5.common import F5ModuleError
+    from library.module_utils.network.f5.common import AnsibleF5Parameters
+    from library.module_utils.network.f5.common import cleanup_tokens
+    from library.module_utils.network.f5.common import fq_name
+    from library.module_utils.network.f5.common import f5_argument_spec
+    from library.module_utils.network.f5.common import flatten_boolean
+    from library.module_utils.network.f5.common import transform_name
+    from library.module_utils.network.f5.common import exit_json
+    from library.module_utils.network.f5.common import fail_json
+except ImportError:
+    from ansible.module_utils.network.f5.bigip import F5RestClient
+    from ansible.module_utils.network.f5.common import F5ModuleError
+    from ansible.module_utils.network.f5.common import AnsibleF5Parameters
+    from ansible.module_utils.network.f5.common import cleanup_tokens
+    from ansible.module_utils.network.f5.common import fq_name
+    from ansible.module_utils.network.f5.common import f5_argument_spec
+    from ansible.module_utils.network.f5.common import flatten_boolean
+    from ansible.module_utils.network.f5.common import transform_name
+    from ansible.module_utils.network.f5.common import exit_json
+    from ansible.module_utils.network.f5.common import fail_json
+
+
+class Parameters(AnsibleF5Parameters):
+    api_map = {
+        'defaultsFrom': 'parent',
+        'insertXforwardedFor': 'insert_xforwarded_for',
+        'redirectRewrite': 'redirect_rewrite',
+        'encryptCookies': 'encrypt_cookies',
+        'encryptCookieSecret': 'encrypt_cookie_secret',
+        'proxyType': 'proxy_type',
+        'explicitProxy': 'explicit_proxy'
```

Comment:
Include a comma here so that we can add or re-organize list items in the future without getting bitten by forgetting to add commas